### PR TITLE
feat: added ReminderModal component

### PR DESCRIPTION
**Pull request by ilasw**
> [!CAUTION]
> Do not merge, I'm using invalid property from `Post`, as added in other RIL PRs

> I didn't bring the logic inside the modal because it is always used inside components where the hook was already called. Maybe a callback `onReminderSet` is enough.

## Changes
- added `ReminderModal` component;
- added utility `useToggle` hook;

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
Web App: https://mi-430-read-it-later-modal.preview.app.daily.dev
Storybook: https://storybook-git-mi-430-read-it-later-modal-dailydotdev.vercel.app/?path=/story/components-notifications-remindermodal--default

### CODE REVIEW

File: packages/shared/src/components/notifications/ReminderModal.tsx
Comment (sshanzel) [inline packages/shared/src/components/notifications/ReminderModal.tsx:16]: Most of the time, we prefer interface over type intersections. More information here: https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections

In summary, extending interfaces is more effective, cached, and displays conflicts.

File: packages/shared/src/components/notifications/ReminderModal.tsx
Comment (sshanzel) [inline packages/shared/src/components/notifications/ReminderModal.tsx:84]: I think we should not need the Fragment here.

**Review by sshanzel — APPROVED** [review]
Looks amazing already to me. Just make sure to verify it on mobile and tablet size as well.

File: packages/shared/src/components/notifications/ReminderModal.tsx
Comment (ilasw) [reply packages/shared/src/components/notifications/ReminderModal.tsx:16]: Cool, thanks for sharing!

File: packages/shared/src/components/notifications/ReminderModal.tsx
Comment (ilasw) [reply packages/shared/src/components/notifications/ReminderModal.tsx:84]: Removing 👍

File: packages/shared/src/components/notifications/ReminderModal.tsx
Comment (sshanzel) [inline packages/shared/src/components/notifications/ReminderModal.tsx:81]: I think this is going to be an issue when using the hook to process the submission. The hook accepts a `preference` prop, and the acceptable values are the labels defined in the enum. So for the `MODAL_OPTIONS` we should also use the `value` as the same as what we put on the label.

Then for the `defaultOption` const, we can then use `defaultOption = ReminderPreference.OneHour` to avoid magic strings.

**Review by sshanzel — CHANGES_REQUESTED** [review]
Just noticed one thing with regards to the data that we will pass. Also, I think we can just define the hook here rather than instantiate it every time we need to open the modal. We can just pass the `post` as a prop to this modal and we should have everything we need to make the hook work.

**Review by nensidosari — COMMENTED** [review]
Looks really good, agree with Lee's comment above as well. 

One thing I noticed but that I think it's more related to the PR that was merged before this is about the hours shown in the modal for selection:
Screenshot 2024-07-11 at 9 38 20 AM

Didn't we want the hours to be 7:00, 9:00 - or we want the mins to be based on the actual time?

File: packages/shared/src/hooks/useToggle.ts
Comment (ilasw) [reply packages/shared/src/hooks/useToggle.ts:7]: I can remove if can be confusing! Just added because I love to use it in different projects 👀

Comment (nensidosari) [reply packages/shared/src/hooks/useToggle.ts:7]: I like it just more of a message to think how we can promote and make it a thing we use

**Comment by ilasw** [comment]
> Didn't we want the hours to be 7:00, 9:00 - or we want the mins to be based on the actual time? cc. @sshanzel

In design is always to 0 minutes, I wrote a function to change it in hook, ready to revert if we don't want it anymore!

**Comment by nensidosari** [comment]
> In design is always to 0 minutes, I wrote a function to change it in hook, ready to revert if we don't want it anymore!


@ilasw u say u changed this so it will be 9:00 now and so on? I don't see that reflected on storybook, still shows as :55 for me based on my time

**Comment by ilasw** [comment]
> @ilasw u say u changed this so it will be 9:00 now and so on? I don't see that reflected on storybook, still shows as :55 for me based on my time

Yes it should work right now, I've to force for storybook redeploy because it is marked as optional and sometime it is skipped :(

File: packages/shared/src/hooks/notifications/useBookmarkReminder.ts
Comment (nensidosari) [inline packages/shared/src/hooks/notifications/useBookmarkReminder.ts:38]: shall we modify the tests for this change?

**Review by nensidosari — APPROVED** [review]
Looks good on my opinion, only the last thing for tests if needed

File: packages/shared/src/hooks/notifications/useBookmarkReminder.ts
Comment (sshanzel) [inline packages/shared/src/hooks/notifications/useBookmarkReminder.ts:38]: Nice. Forgot to consider the minutes. Good catch!

**Review by sshanzel — APPROVED** [review]
Looks amazing! Great fix on the minute-level of calculation 🚀

File: packages/shared/src/hooks/notifications/useBookmarkReminder.ts
Comment (sshanzel) [reply packages/shared/src/hooks/notifications/useBookmarkReminder.ts:38]: Yep. Should be part of the tests to ensure accurate result.

Comment (ilasw) [reply packages/shared/src/hooks/notifications/useBookmarkReminder.ts:38]: Done ✅

**Review by rebelchris — APPROVED** [review]
Thanks for keeping stories going, I love it!
Non blocking feedback from my side.

File: packages/shared/src/components/notifications/ReminderModal.tsx
Comment (ilasw) [reply packages/shared/src/components/notifications/ReminderModal.tsx:65]: It behaves like a radio button so this role should improve accessibility, I would like to use the radio group but it was too much styling :\

File: packages/shared/src/components/notifications/ReminderModal.tsx
Comment (ilasw) [reply packages/shared/src/components/notifications/ReminderModal.tsx:104]: There is another way of doing this? Classes cannot be override otherwise... We should use that `twMerge` 🤔

Comment (rebelchris) [reply packages/shared/src/components/notifications/ReminderModal.tsx:103]: No i mean why are they different from our guideline modals more 🙏

File: packages/shared/src/components/notifications/ReminderModal.tsx
Comment (sshanzel) [reply packages/shared/src/components/notifications/ReminderModal.tsx:110]: I think this one is for Desktop. Does it not display the same header on mobile without the `title` prop? If so we might really need it.

Comment (ilasw) [reply packages/shared/src/components/notifications/ReminderModal.tsx:110]: @sshanzel this ModalHeader will not show at mobile breakpoints, only adding title to `drawerProps` will make it works.

**Comment by nensidosari** [comment]
@ilasw I see your description to not merge this PR, if still valid maybe consider pointing this PR to this feature branch [MI-432](https://github.com/dailydotdev/apps/tree/MI-432) so everything about bookmarks can be there before we merge the final PR

File: packages/shared/src/components/notifications/ReminderModal.tsx
Comment (sshanzel) [reply packages/shared/src/components/notifications/ReminderModal.tsx:110]: I see. Then we really need it in this case.

**Pull request merged**
